src/scripts/main.js

```diff
@@ -1,3 +1,251 @@
+`;
+
+place.append(panel);
+
+function pushNotification(posTop, posRight, title, description, type = '') {
+  const mesage = document.createElement('div');
+
+  mesage.className = `notification ${type}`;
+  mesage.dataset.qa = 'notification';
+  mesage.style.top = `${posTop}px`;
+  mesage.style.right = `${posRight}px`;
+
+  mesage.innerHTML = `
+    <h2 class="title">${title}</h2>
+    <p>${description}</p>
+  `;
+
+  place.append(mesage);
+
+  setTimeout(() => {
+    mesage.hidden = true;
+  }, 2000);
+}
+
+function checkData(formData) {
+  if (formData.name.length < 4) {
+    pushNotification(500, 10, 'Error!',
+      `Something went wrong!<br> `
+      + `Name field cannot be empty! Or less then 4 letters!`, 'error');
+
+    return false;
+  }
+
+  if (!formData.position) {
+    pushNotification(500, 10, 'Error!',
+      `Something went wrong!<br> `
+      + `Position field cannot be empty!`, 'error');
+
+    return false;
+  }
+
+  if (+formData.age < 18 || +formData.age > 90) {
+    pushNotification(500, 10, 'Error!',
+      `Something went wrong!<br> `
+      + `Age must be more 18 & less 90`, 'error');
+
+    return false;
+  }
+
+  return true;
+}
+
+const listValue = document.querySelector('tbody');
+
+panel.addEventListener('submit', e => {
+  e.preventDefault();
+
+  const data = Object.fromEntries(new FormData(panel).entries());
+
+  const newRow = document.createElement('tr');
+
+  const valueSalary = Number(data.salary).toLocaleString('en-US');
+
+  newRow.innerHTML = `
+    <td>${data.name}</td>
+    <td>${data.position}</td>
+    <td>${data.office}</td>
+    <td>${data.age}</td>
+    <td>${'$' + valueSalary}</td>
+  `;
+
+  if (checkData(data) === true) {
+    pushNotification(500, 10, 'Success!',
+      'Good!\n '
+    + 'Employee has been added!', 'success');
+
+    listValue.append(newRow);
+    panel.reset();
+  } else {
+    checkData(data);
+  }
+});
+
+function convertCurrencyToNumber(value) {
+  return +value.slice(1).split(',').join('');
+}
+
+function sortList(list, index, sortMethod) {
+  const result = [...list.children].sort((a, b) => {
+    if (index === 4) {
+      const valueA = convertCurrencyToNumber(a.children[index].innerText);
+      const valueB = convertCurrencyToNumber(b.children[index].innerText);
+
+      if (sortMethod === 'down') {
+        return valueB - valueA;
+      }
+
+      return valueA - valueB;
+    }
+
+    const value1 = a.children[index].innerText;
+    const value2 = b.children[index].innerText;
+
+    if (sortMethod === 'down') {
+      return value2.localeCompare(value1);
+    }
+
+    return value1.localeCompare(value2);
+  });
+
+  list.append(...result);
+}
+
+addEventListener('click', e => {
+  const tableHeaders = document.querySelector('thead').children[0].children;
+  const tableFooter = document.querySelector('tfoot');
+  const i = [...tableHeaders].findIndex(
+    element => element.innerText === e.target.innerText
+    && !tableFooter.contains(e.target));
```

Comment:
don't see any sense for creating copy of array here
```suggestion
  const i = tableHeaders.findIndex(
    element => element.innerText === e.target.innerText
    && !tableFooter.contains(e.target));
```